webapp/src/components/add_issue/add_issue.jsx

```diff
@@ -0,0 +1,330 @@
+
+import {
+    makeStyleFromTheme,
+    changeOpacity,
+} from 'mattermost-redux/utils/theme_utils';
+
+import TextareaAutosize from 'react-textarea-autosize';
+
+import FullScreenModal from '../modals/modals.jsx';
+import Button from '../../widget/buttons/button';
+import Chip from '../../widget/chip/chip';
+import AutocompleteSelector from '../user_selector/autocomplete_selector.tsx';
+import './add_issue.scss';
+import CompassIcon from '../icons/compassIcons';
+import { getProfilePicture } from '../../utils';
+
+const PostUtils = window.PostUtils;
+
+export default class AddIssue extends React.Component {
+    static propTypes = {
+        visible: PropTypes.bool.isRequired,
+        message: PropTypes.string.isRequired,
+        postID: PropTypes.string.isRequired,
+        assignee: PropTypes.object,
+        closeAddBox: PropTypes.func.isRequired,
+        submit: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        autocompleteUsers: PropTypes.func.isRequired,
+        openAssigneeModal: PropTypes.func.isRequired,
+        removeAssignee: PropTypes.func.isRequired,
+    };
+
+    shouldComponentUpdate() {
+        return true;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            message: '',
+            description: '',
+            sendTo: null,
+            attachToThread: false,
+            previewMarkdown: false,
+            assigneeModal: false,
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (props.visible && state.message == null) {
+            return { message: props.message };
+        }
+        if (!props.visible && (state.message != null || state.sendTo != null)) {
+            return {
+                message: null,
+                sendTo: null,
+                attachToThread: false,
+                previewMarkdown: false,
+            };
+        }
+        return null;
+    }
+
+    handleAttachChange = (e) => {
+        const value = e.target.checked;
+        if (value !== this.state.attachToThread) {
+            this.setState({
+                attachToThread: value,
+            });
+        }
+    };
+
+    close = () => {
+        const { closeAddBox, removeAssignee } = this.props;
+        removeAssignee();
+        closeAddBox();
+    }
+
+    submit = () => {
+        const { submit, postID, assignee, closeAddBox, removeAssignee } = this.props;
+        const { message, description, attachToThread } = this.state;
+        this.setState({
+            message: '',
+            description: '',
+        });
+        if (attachToThread) {
+            submit(message, description, assignee.username, postID);
+        } else {
+            submit(message, description);
+        }
+
+        removeAssignee();
+        closeAddBox();
+    };
+
+    toggleAssigneeModal = (value) => {
+        this.setState({ assigneeModal: value });
+    }
+
+    onKeyDown = (e) => {
+        if (e.key === 'Enter') {
```

Comment:
Do you want to consider things like cmd+enter to add a new line?

Reply:
Not at this point. Maybe as the next iteration we can add it in.